src/runtime/gc_stack_raw.go

```diff
@@ -3,12 +3,38 @@
 
 package runtime
 
+import "internal/task"
+
 // markStack marks all root pointers found on the stack.
 //
 // This implementation is conservative and relies on the stack top (provided by
 // the linker) and getting the current stack pointer from a register. Also, it
 // assumes a descending stack. Thus, it is not very portable.
 func markStack() {
-	// Mark system stack.
-	markRoots(getSystemStackPointer(), stackTop)
+	// Scan the current stack, and all current registers.
+	localscan()
+
+	if !task.OnSystemStack() {
+		// Mark system stack.
+		markRoots(getSystemStackPointer(), stackTop)
+	}
+}
+
+//go:export tinygo_localscan
+func localscan()
```

Comment:
I think `localscan` is not very descriptive. What about `scanstackwrapper`, `markStackWrapper`, `markRegistersAndStack`?